### PR TITLE
Adds Github Action Support

### LINKED ISSUE #5: Plugin Not running on github action
The plugin is reaching the maximum CPU wall time on Cloudflare. So, instead of workers it should run on actions.

### PR DESCRIPTION
**Pull request by shiv810**
Resolves #5 

- Added compute.yml for github actions
- Fixed types

### CODE REVIEW

File: .github/workflows/compute.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        description: "Event Name"
+      eventPayload:
+        description: "Event Payload"
+      settings:
+        description: "Settings"
+      authToken:
+        description: "Auth Token"
+      ref:
+        description: "Ref"
+
```
Comment (gentlementlegen) [inline .github/workflows/compute.yml:18]:
```suggestion
signature:
    description: "Signature sent from the Kernel"
```

File: .github/workflows/update-configuration.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  update-manifest:
+    name: "Update Manifest and Commit Changes"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update Manifest and Commit Changes
+        uses: ubiquity-os/action-deploy-plugin@main
+        with:
+          manifestPath: ${{ github.workspace }}/manifest.json
+          schemaPath: ${{ github.workspace }}/src/types/plugin-input.js
```
Comment (gentlementlegen) [inline .github/workflows/update-configuration.yml:17]:
```suggestion
          schemaPath: ${{ github.workspace }}/src/types/plugin-input.ts
```

File: src/helpers/issue.ts [hunk 3]
```diff
@@ -163,7 +162,9 @@ export async function fetchCodeLinkedFromIssue(
       repo: context.payload.repository.name,
       issueNumber: parseInt(issue.match(/\/issues\/(\d+)/)?.[1] || "0", 10),
       issueUrl: url,
-      user: null,
+      user: {
```
Comment (gentlementlegen) [inline src/helpers/issue.ts:165]:
```suggestion
      user: context.payload.sender
```

File: src/helpers/format-chat-history.ts [hunk 4]
```diff
@@ -140,9 +140,9 @@ function createComment(comment: StreamlinedComments) {
     return "";
   }
   // Remove duplicates
-  const uniqueComments = comment.comments.filter((c, i, a) => a.findIndex((cc) => cc.id === c.id) === i);
+  //const uniqueComments = comment.comments.filter((c, i, a) => a.findIndex((cc) => cc.id === c.id) === i);
```
Comment (0x4007) [inline src/helpers/format-chat-history.ts:143]:
```suggestion
```

File: .github/workflows/compute.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+name: "@ubiquity-os/command-ask"
```
Comment (0x4007) [inline .github/workflows/compute.yml:1]:
```suggestion
name: "@ubiquity-os-marketplace/command-ask"
```

File: .github/workflows/update-configuration.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+name: Update Manifest and Commit Changes
+
+on:
+  workflow_dispatch:
+  push:
+
+
```
Comment (gentlementlegen) [inline .github/workflows/update-configuration.yml:7]:
```suggestion
```

File: .github/workflows/compute.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+jobs:
+  compute:
+    name: "LLM Reply"
+    runs-on: ubuntu-latest
+    permissions: write-all
+    env:
+      SUPABASE_URL: ${{ secrets.SUPABASE_URL }}
+      SUPABASE_KEY: ${{ secrets.SUPABASE_KEY }}
+      VOYAGEAI_API_KEY: ${{secrets.VOYAGEAI_API_KEY}}
```
Comment (gentlementlegen) [inline .github/workflows/compute.yml:29]:
```suggestion
      VOYAGEAI_API_KEY: ${{ secrets.VOYAGEAI_API_KEY }}
```
(applies to all the secrets below)

File: .github/workflows/compute.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      - name: setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.10.0"
+
+      - name: install dependencies
+        run: yarn
+
+      - name: execute directive
+        run: npx tsx ./src/main.ts
```
Comment (gentlementlegen) [inline .github/workflows/compute.yml:45]:
```suggestion
        run: yarn tsx ./src/main.ts
```

Comment (gentlementlegen) [reply .github/workflows/compute.yml]: If you are using both indeed maybe not. I think you can include that file within `knip` configuration instead, in the `entry` section
https://knip.dev/overview/configuration#defaults

**Review by gentlementlegen — APPROVED** [review]

File: .github/workflows/compute.yml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      eventName:
+        description: "Event Name"
+      eventPayload:
+        description: "Event Payload"
+      settings:
+        description: "Settings"
+      authToken:
+        description: "Auth Token"
+      ref:
+        description: "Ref"
```
Comment (0x4007) [inline .github/workflows/compute.yml:17]: This should have an explanation it's unclear what this is

**Review by 0x4007 — APPROVED** [review]

**Pull request merged**